app/src/main/java/org/fossasia/openevent/general/auth/AuthService.kt

```diff
@@ -33,39 +30,31 @@ class AuthService(private val authApi: AuthApi,
 
     fun isLoggedIn() = authHolder.isLoggedIn()
 
-    fun logout() {
-        val userFlowable = userDao.getUser(authHolder.getId())
-        userFlowable.map {
-            Single.fromCallable {
+    fun logout(): Completable {
+        return Completable.fromAction({
+            var userFlowable = userDao.getUser(authHolder.getId())
+            userFlowable.map {
                 userDao.deleteUser(it)
-            }.subscribeOn(Schedulers.io())
-                    .observeOn(AndroidSchedulers.mainThread())
-                    .subscribe({
-                        Timber.d("User logged out!")
-                    }
-                    ) {
-                        Timber.e(it, "Error logging out user!")
-                    }
-        }
-        authHolder.token = null
+            }
```

Comment:
Won't work as there is nobody subscribing it

Reply:
should I then keep it inside above map statement?

Reply:
You can